internal/rbd/nodeserver.go

```diff
@@ -291,13 +273,19 @@ func (ns *NodeServer) NodeStageVolume(
 	defer ns.VolumeLocks.Release(volID)
 
 	// Check this is a migration request because in that case, unlike other node stage requests
-	// it will be missing the clusterID, so fill it by fetching it from config file using mon.
-	if req.GetVolumeContext()[intreeMigrationKey] == intreeMigrationLabel && req.VolumeContext[util.ClusterIDKey] == "" {
-		cID, cErr := getClusterIDFromMigrationVolume(req.GetVolumeContext()["monitors"])
-		if cErr != nil {
-			return nil, status.Error(codes.Internal, cErr.Error())
+	// it will be missing the clusterID, so fill it by fetching it from config file using monitors hash.
+	if isMigrationContext(req.GetVolumeContext()) {
+		// verify and parse volID
+		if isMigrationVolID(volID) {
+			parsedMigHandle, pErr := parseMigrationVolID(volID)
+			if pErr != nil {
+				log.ErrorLog(ctx, "failed to parse the migration handle, err :%v ", pErr)
```

Comment:
just use `failed to parse the migration handle: %v` as error format, the last component after `:` is expected to be the error message

Reply:
done